### ISSUE TITLE
ExtUtils::ParseXS fails basic.t test without dynamic loading

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#9737]
Migrated from [rt.perl.org#65754](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=65754) (status was 'resolved')

Searchable as RT65754$

**Comment by p5pRT** [comment]
### From @doughera88

#### Created by @doughera88



When built without dynamic loading \(e\.g\. Configure \-Uusedl\)\, tests 6\-10 
of ExtUtils/ParseXS/t/basic\.t fail\.  I have attached the result of
running

&nbsp;     cd t; \./perl harness \-v \.\./lib/ExtUtils/ParseXS/t/basic\.t 2>&1

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=low
    module=ExtUtils::ParseXS

Site configuration information for perl 5.11.0:

Configured by doughera at Mon May 18 09:23:18 EDT 2009.

Summary of my perl5 (revision 5 version 11 subversion 0 patch GitLive-blead-1145-gb3a120b) configuration:
  Snapshot of: b3a120bf72d91507f4a2782bf0be31aa8b4dba31
  Platform:
    osname=linux, osvers=2.6.26-2-686, archname=i686-linux
    uname='linux fractal 2.6.26-2-686 #1 smp thu mar 26 01:08:11 utc 2009 i686 gnulinux '
    config_args='-Uusedl -Dprefix=/opt/perl -Doptimize=none -des -Dusedevel'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=undef, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize=' ',
    cppflags='-fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.3.2', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='ld', ldflags =' -fstack-protector -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib /usr/lib64
    libs=-lnsl -lgdbm -ldl -lm -lcrypt -lutil -lc -lgdbm_compat
    perllibs=-lnsl -lgdbm -ldl -lm -lcrypt -lutil -lc -lgdbm_compat
    libc=/lib/libc-2.7.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.7'
  Dynamic Linking:
    dlsrc=dl_none.xs, dlext=none, d_dlsymun=undef, ccdlflags=''
    cccdlflags='', lddlflags=''

Locally applied patches:
    PERL_GIT_UNPUSHED_COMMITS    	/* do not remove this line */
    PERL_GIT_UNCOMMITTED_CHANGES	/* do not remove this line */


@INC for perl 5.11.0:
    lib
    /opt/perl/lib/site_perl/5.11.0/i686-linux
    /opt/perl/lib/site_perl/5.11.0
    /opt/perl/lib/5.11.0/i686-linux
    /opt/perl/lib/5.11.0
    /opt/perl/lib/site_perl
    .


Environment for perl 5.11.0:
    HOME=/home/doughera
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH=/usr/local/lib
    LOGDIR (unset)
    PATH=/home/doughera/script:/home/doughera/bin:/opt/perl/script:/opt/perl/bin:/opt/dist/bin:/opt/gnuplot/bin:/opt/groff/bin:/usr/local/script:/usr/local/bin:/opt/gnu/script:/opt/gnu/bin:/opt/dist/script:/bin:/usr/bin:/usr/X11R6/bin:/usr/openwin/bin:.
    PERL_BADLANG (unset)
    SHELL=/bin/bash

-- 
    Andy Dougherty		doughera@lafayette.edu
    Dept. of Physics
    Lafayette College, Easton PA 18042

```
</details>

**Comment by p5pRT** [comment]
### From @nwc10

Dave notes&#8203;:

Andy D pointed this out \- I don't know whether its a regression

**Comment by p5pRT** [comment]
@nwc10 - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From alex@chmrr.net

On Thu May 28 08&#8203;:25&#8203;:43 2009\, nicholas wrote&#8203;:
> Andy D pointed this out \- I don't know whether its a regression

This is \_not\_ a regression \-\- it also fails on 5\.8\.8 with \-Uusedl\. 
Because I don't see the log file which was nominally attached to the
original report\, bleadperl's results are attached\.
 \- Alex

**Comment by p5pRT** [comment]
### From alex@chmrr.net

[ExtUtils-ParseXS-Uusedl.log](https://rt-archive.perl.org/perl5/Ticket/Attachment/569908/270782/ExtUtils-ParseXS-Uusedl.log)

**Comment by p5pRT** [comment]
### From p5p@spam.wizbit.be

On Sun May 31 00&#8203;:04&#8203;:36 2009\, chmrr wrote&#8203;:
> On Thu May 28 08&#8203;:25&#8203;:43 2009\, nicholas wrote&#8203;:
> > Andy D pointed this out \- I don't know whether its a regression
> 
> This is \_not\_ a regression \-\- it also fails on 5\.8\.8 with \-Uusedl\. 
> Because I don't see the log file which was nominally attached to the
> original report\, bleadperl's results are attached\.
>  \- Alex
> 


This is indeed not a regression\.

ExtUtils/ParseXS was first included in Change 24500; attempting to 
build it and running the tests with \-Uusedl fails\.

So it appears that building/testing ExtUtils/ParseXS on a perl compiled 
with \-Uusedl has never worked\.\.\.

**Comment by p5pRT** [comment]
### From @xdg

ExtUtils&#8203;::ParseXS 2\.19\_04 will skip sections of tests if $Config\{usedl\}
is false\.

Patch submitted to perl5\-porters&#8203;:

http&#8203;://www\.xray\.mpe\.mpg\.de/mailing\-lists/perl5\-porters/2009\-06/msg01040\.html

\-\- David

**Comment by p5pRT** [comment]
### From @xdg

Fixed in CPAN ExtUtils&#8203;::ParseXS and merged into blead and maint\.

**Comment by p5pRT** [comment]
@xdg - Status changed from 'open' to 'resolved'
